Ain't That a Kick in the Head

The32bitguy opened the pull request:
# About the pull request

Sangeili AI can now kick.
AI with TRAIT_SUPER_STRONG now attempts grabs and throw.
more to come
# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

https://github.com/user-attachments/assets/9144f335-efd2-4688-9b19-ff97082c73f4


Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>


# Changelog
:cl:
add: Sangeili AI can now kick.
add: AI with TRAIT_SUPER_STRONG now wrestle.
/:cl:

The32bitguy committed:
initial

cm13-github commented:
<!-- tgs_test_merge_comment -->
## Test merge deployment history:

<details><summary>Test Merge Deployed @ 10/03/2025 03:46:59 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 5f7e7f63c2e6712e5183435ec630f2183f5b9fcb
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: e9c52cf587e653f0ef3ea3bdadafe829d3f917ff
</details>
<details><summary>Test Merge Updated @ 10/03/2025 04:22:00 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 5f7e7f63c2e6712e5183435ec630f2183f5b9fcb
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: f96d996d4b889b5825000335daa3aedf48953b77
</details>
<details><summary>Test Merge Updated @ 10/30/2025 02:15:20 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: beda11cae69f210127545a2f4f501506d3cda149
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: c6b599c88f3c2dfb39cd0226207c83aa83c86b2c
</details>
<details><summary>Test Merge Updated @ 10/30/2025 23:13:43 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: beda11cae69f210127545a2f4f501506d3cda149
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: d83465ecda946c426807339a8bb84714d244a4e8
</details>
<details><summary>Test Merge Updated @ 11/01/2025 00:01:23 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: beda11cae69f210127545a2f4f501506d3cda149
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 3665fffb206b26be38bc6f69e3b115396804a837
</details>
<details><summary>Test Merge Updated @ 11/06/2025 19:23:20 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 285e23a5231bcfe8d9fe97135ad724e48c69c99a
</details>
<details><summary>Test Merge Updated @ 11/07/2025 05:38:49 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
AndroBetel

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: f8ef562e62917d75b971189a7aea59e24e0446f3
</details>
<details><summary>Test Merge Updated @ 11/07/2025 06:19:40 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Athena14

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 5ebdf888b8019ebabe2d7bc42018571da51e25e8
</details>
<details><summary>Test Merge Updated @ 11/07/2025 07:10:26 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: c1133db7a4e24b6c226e4936afef738c57d0bee2
</details>
<details><summary>Test Merge Updated @ 11/07/2025 08:32:15 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 971838312de5febf347796effe47213bcf4e1e3e
</details>
<details><summary>Test Merge Updated @ 11/07/2025 08:39:59 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: e97abd2e0a5023c618bcc425f47bbbbaa03874be
</details>
<details><summary>Test Merge Updated @ 11/07/2025 08:43:02 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: eeef92950d4c3a2073abc40e7430757d3204b3f2
</details>
<details><summary>Test Merge Updated @ 11/14/2025 08:02:49 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Some Random Username

##### Revision
Origin: d78e62e9ee370988ec84f5741f00a56900afbbbf
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 2fe720365383a29a8a43ed7282cff1a9c9b4b23c
</details>
<details><summary>Test Merge Removed @ 11/14/2025 08:09:33 +00:00:</summary>

##### Server Instance
PVECM: Halo
</details>
<details><summary>Test Merge Deployed @ 11/24/2025 04:23:10 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: eef640ca2d0182de086a807931f7759e9803f12b
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 3d49097c49d21d8166fb66699f22ea9d8045bce6
</details>
<details><summary>Test Merge Updated @ 11/24/2025 22:47:17 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 9ddc56c232ce5295609742141c865283edfdbeb9
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: fb99056ce88b92c52753de4b1798319bd0be77ea
</details>
<details><summary>Test Merge Removed @ 11/27/2025 02:50:56 +00:00:</summary>

##### Server Instance
PVECM: Halo
</details>
<details><summary>Test Merge Deployed @ 11/27/2025 02:55:00 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 44b6d3d8cef69b2c7b7a694787291fea3824cfdd
</details>
<details><summary>Test Merge Updated @ 11/27/2025 02:57:33 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: c8463556588cd83b679a0df8ad0d2dc86976c356
</details>
<details><summary>Test Merge Updated @ 11/28/2025 22:35:33 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: d4ad3fa2dc6ede4a41f405012ac09c53d427e750
</details>
<details><summary>Test Merge Updated @ 11/28/2025 22:48:37 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: b6dc8278e554a45a148e05f462ac6e3fa3dab430
</details>
<details><summary>Test Merge Updated @ 11/28/2025 22:50:07 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 43b5c7bbdd09e55429141ae5b89515617e473af7
</details>
<details><summary>Test Merge Updated @ 11/28/2025 22:59:38 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: e9a22ac83f27c4d26283b50a1a12b158251af91a
</details>
<details><summary>Test Merge Updated @ 11/28/2025 23:05:14 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 7feac9b77a753051a5159a9a69129edc97ac5e85
</details>
<details><summary>Test Merge Updated @ 11/28/2025 23:14:11 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 33fb2653577836c80e4217880f72cccb828b5e1d
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 7af5ba6544e601ab8a1b7b8ad3190392d12f95b6
</details>
<details><summary>Test Merge Updated @ 12/14/2025 10:08:44 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
gdspathe

##### Revision
Origin: 8bd66ce97131d6fce2c9916b6810ac356f5996ff
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 3bec5e217a3b8eef2cc19a6e889726c8cd4b80d9
</details>
<details><summary>Test Merge Updated @ 12/14/2025 10:45:39 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
gdspathe

##### Revision
Origin: 8bd66ce97131d6fce2c9916b6810ac356f5996ff
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 12ef35e6ea473bc935340bde0618ff66fc1559f5
</details>
<details><summary>Test Merge Updated @ 12/20/2025 07:51:18 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
SpartanBobby

##### Revision
Origin: 8bd66ce97131d6fce2c9916b6810ac356f5996ff
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: d2de8b0c416fb2e52525c0c89115787014a0dc04
</details>
<details><summary>Test Merge Updated @ 12/22/2025 05:22:23 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 8bd66ce97131d6fce2c9916b6810ac356f5996ff
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: f2aeee08284554e90bb4db9a25cd37d18f9b7214
</details>
<details><summary>Test Merge Updated @ 12/22/2025 21:49:45 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
tristan63

##### Revision
Origin: 8bd66ce97131d6fce2c9916b6810ac356f5996ff
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: a4943b3f4d64081ad77ca8385216f8e457156e25
</details>
<details><summary>Test Merge Updated @ 12/23/2025 11:38:20 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 834cb221cc255973dac8c6094674316f05d90c65
</details>
<details><summary>Test Merge Updated @ 12/23/2025 11:48:06 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 7e6d68200bec2ccf0809f10b049af8d729522d11
</details>
<details><summary>Test Merge Updated @ 12/23/2025 12:09:36 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: af425d324bc2eb35c9b9696e6db2db7483137685
</details>
<details><summary>Test Merge Updated @ 12/23/2025 12:15:01 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: e063fd810fd9d9155d81622051cf43c23a94667b
</details>
<details><summary>Test Merge Updated @ 12/23/2025 12:24:02 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 9588d870f44b3d0408befd34cd973caa7d6a77e4
</details>
<details><summary>Test Merge Updated @ 12/23/2025 12:39:15 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 1b974777c2ca810a343a0491b5ed2b4306bc6af8
</details>
<details><summary>Test Merge Updated @ 12/23/2025 12:54:35 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 403bc2268a452cb37af60aeaa58616808baf962c
</details>
<details><summary>Test Merge Updated @ 12/23/2025 13:10:52 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: c6f4eaf59f88fdb89548d8df9722f21341f949b2
</details>
<details><summary>Test Merge Updated @ 12/24/2025 13:55:23 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 138e9df23109e13feeab3b4b77bc7f4e213e512e
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 032dd6d3869986130b163d1c8fdb9170e05fdaad
</details>
<details><summary>Test Merge Removed @ 01/05/2026 11:48:05 +00:00:</summary>

##### Server Instance
PVECM: Halo
</details>
<details><summary>Test Merge Deployed @ 01/05/2026 12:33:24 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 91d656cef0978d73de6537e7fe6e9096b5067592
</details>
<details><summary>Test Merge Updated @ 01/10/2026 08:04:24 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: d9b611d9d20ada4c22aea04cf402e441e168796e
</details>
<details><summary>Test Merge Updated @ 01/10/2026 08:19:25 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 2c5079acacb923e778a5372fad00c90b57bc145e
</details>
<details><summary>Test Merge Updated @ 01/13/2026 07:26:58 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 204143af47e552f70c0cd6bc3f6e6f5d0931f3cf
</details>
<details><summary>Test Merge Updated @ 01/14/2026 00:05:47 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 7e5737af213764a8060b766e2512f30ca93e7699
</details>
<details><summary>Test Merge Updated @ 01/16/2026 22:33:11 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: b3bf7872838876ba6b5d115b0bc46265a7c6d99e
</details>
<details><summary>Test Merge Updated @ 01/16/2026 23:04:25 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 76a31f6c65915b2b52eee3a7b0f34bfeb034717f
</details>
<details><summary>Test Merge Updated @ 01/16/2026 23:26:59 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: b608862f2f8b74bc4fddf1de251c3240d376c557
</details>
<details><summary>Test Merge Updated @ 01/16/2026 23:40:06 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: bf8243288b60ce35c415ae1889dd5d8920fa92cf
</details>
<details><summary>Test Merge Updated @ 01/17/2026 03:29:11 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 4f16f5e9bb109ce2a15c0b7c78c9e32c1ffe7648
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: b90a990f82a091e2b69b0e24e5523653e316218f
</details>
<details><summary>Test Merge Updated @ 01/17/2026 10:47:30 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 1bea42ec97bf17860b1bef06f65d778618145bf8
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 234f76e3329ec9d6761725b64ed176433bd7fed9
</details>
<details><summary>Test Merge Updated @ 01/17/2026 11:41:10 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 1bea42ec97bf17860b1bef06f65d778618145bf8
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 04d875432299a48af6a8c87d3b8568cf3578d6f6
</details>
<details><summary>Test Merge Updated @ 01/18/2026 07:45:10 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ody_sseus

##### Revision
Origin: dc38691ffb7928cd8c891c14626b73f572981a47
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 5363a5be531d7fa0205615287051a6e9ab437e5d
</details>
<details><summary>Test Merge Updated @ 01/18/2026 08:27:24 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ody_sseus

##### Revision
Origin: dc38691ffb7928cd8c891c14626b73f572981a47
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: e905ade9706598462704dd52aeb5400fa7807748
</details>
<details><summary>Test Merge Updated @ 01/18/2026 08:34:39 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ody_sseus

##### Revision
Origin: dc38691ffb7928cd8c891c14626b73f572981a47
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 898d0500ac60fa8a20a40744dbdd9f611ffaf06b
</details>
<details><summary>Test Merge Updated @ 01/18/2026 08:40:48 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ody_sseus

##### Revision
Origin: dc38691ffb7928cd8c891c14626b73f572981a47
Pull Request: 55dbde031ec7211fa3c7dc7a46525ba8337b96b4
Server: 798ccc148330978ad7848209ba955b473ace44f9
</details>
<details><summary>Test Merge Updated @ 01/19/2026 02:54:17 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 93bf5a06de25a87d4f830b61b9888a4ea62350f2
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: a21a5676fdea8c48daef477c91f3fb64fab3e21f
</details>
<details><summary>Test Merge Updated @ 01/21/2026 10:12:20 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 93bf5a06de25a87d4f830b61b9888a4ea62350f2
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 2dd82dfca322d9dfbb1e70c5840547c486575e41
</details>
<details><summary>Test Merge Updated @ 01/25/2026 04:43:27 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 93bf5a06de25a87d4f830b61b9888a4ea62350f2
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 4dab8d36ae87d680182e0ed9ae13223f1d8e636a
</details>
<details><summary>Test Merge Updated @ 02/03/2026 18:24:03 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: f227ef281c6580e26e665bc191f77e4cc2c2b986
</details>
<details><summary>Test Merge Updated @ 02/05/2026 01:59:40 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 4921b6831a836120ee976c537d3a5f6c688fd31d
</details>
<details><summary>Test Merge Updated @ 02/06/2026 17:43:48 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: d80bb0b06986f3530ac3d25d48eb67ff4fbf3ddc
</details>
<details><summary>Test Merge Updated @ 02/06/2026 18:04:47 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 60820ee430721e0a9d69bf0172327f78986543b4
</details>
<details><summary>Test Merge Updated @ 02/06/2026 18:06:54 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: d80bb0b06986f3530ac3d25d48eb67ff4fbf3ddc
</details>
<details><summary>Test Merge Removed @ 02/06/2026 18:15:24 +00:00:</summary>

##### Server Instance
PVECM: Halo
</details>
<details><summary>Test Merge Deployed @ 02/06/2026 19:31:37 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Verbs

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 4921b6831a836120ee976c537d3a5f6c688fd31d
</details>
<details><summary>Test Merge Updated @ 02/16/2026 02:53:19 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
banana-battery

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 892daa1f84cc62c0ce60554a1fe128a448cd720c
</details>
<details><summary>Test Merge Updated @ 02/17/2026 01:52:59 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ShortTemperedLeprech

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: f5ed50ae8b0da31f11e709b86107d5db3d7cf425
</details>
<details><summary>Test Merge Updated @ 02/21/2026 01:34:59 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
ShortTemperedLeprech

##### Revision
Origin: 752f940da43669b62e483ce4451deecfaf4de064
Pull Request: 19b734f301bac4919bcd07d8c230eb75239e047a
Server: 28b0e60e1d07a46a7270344621fc821d5378b15b
</details>
<details><summary>Test Merge Removed @ 02/24/2026 23:15:10 +00:00:</summary>

##### Server Instance
PVECM: Halo
</details>
<details><summary>Test Merge Deployed @ 05/03/2026 06:29:37 +00:00:</summary>

##### Server Instance
PVECM: Halo

##### Merged By
Karmac

##### Revision
Origin: 9a9285051c8cee1594577864c3f4942dea8a0aad
Pull Request: 55eba61e67b88d245e01066c116689bf3c1e0727
Server: 59403b28ccb20bf7444e2a16322e11993cede54e
</details>

The32bitguy committed:
s

The32bitguy committed:
Merge remote-tracking branch 'upstream/master' into cov-cqc

The32bitguy committed:
bug fix

The32bitguy committed:
Merge remote-tracking branch 'upstream/master' into cov-cqc

The32bitguy committed:
fix indentation

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/halo/kick.dm:
```diff
@@ -0,0 +1,39 @@
+	if(!brain.move_to_next_turf(get_turf(brain.current_target)))
+		return ONGOING_ACTION_COMPLETED
+
+	return ONGOING_ACTION_COMPLETED
```
you don't need the if() there

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/halo/kick.dm:
```diff
@@ -0,0 +1,39 @@
+
+	var/mob/tied_human = brain.tied_human
+	if(get_dist(tied_human, brain.current_target) <= 1)
+		if(prob(70))
```
why the prob 70?

The32bitguy replied:
magic number

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/halo/kick.dm:
```diff
@@ -0,0 +1,39 @@
+	var/mob/tied_human = brain.tied_human
+	if(get_dist(tied_human, brain.current_target) <= 1)
+		if(prob(70))
+			//tied_human.a_intent_change(INTENT_HARM)
```
?

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/fire_at_target.dm:
```diff
@@ -31,6 +31,11 @@
 	if(brain.should_reload())
 		return 0
 
+	if(istype(brain.primary_weapon, /obj/item/weapon/gun/energy/plasma))
```
any way this could put in should_reload() instead? I forget what that proc looks like these days

Contrabang commented on code/modules/covenant/species/sangheili/sangheili_actions.dm:
```diff
@@ -71,7 +71,8 @@
 
 /datum/action/human_action/activable/covenant/sangheili_kick/use_ability(atom/target, mob/living/carbon/owner)
-	owner = usr
+	if(owner == null)
```
I would suggest using any falsey value, just to be safe
```suggestion
	if(!owner)
```

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/halo/kick.dm:
```diff
@@ -0,0 +1,39 @@
+/datum/ai_action/kick
```
does this need action flags

The32bitguy replied:
It is more of a limbless magic ability than what seems to be against the precident for what needs action_flags.

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/fire_at_target.dm:
```diff
@@ -230,7 +235,25 @@
+				currently_firing = FALSE
+				addtimer(CALLBACK(brain.primary_weapon, TYPE_PROC_REF(/obj/item/weapon/gun, start_fire), null, brain.current_target, null, null, null, TRUE), brain.primary_weapon.get_fire_delay())
+				return
+		addtimer(CALLBACK(plasma_reducer, TYPE_PROC_REF(/obj/item/weapon/gun/energy/plasma/plasma_pistol, start_fire), tied_human), plasma_reducer.get_fire_delay()*0.25) //standard hAi weapon throttling
```
isn't fire delay * 0.25 making it fire 4x as fast?

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/action_datums/walk_melee.dm:
```diff
@@ -34,12 +34,26 @@
 		return ONGOING_ACTION_COMPLETED
 
-	return ONGOING_ACTION_COMPLETED
+	return ONGOING_ACTION_UNFINISHED
```
why this change?

Contrabang commented on code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm:
```diff
@@ -136,6 +136,10 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
+	for(var/datum/ai_action/act in allowed_actions)
+		if(act.species_restricted)
+			if(act.species_restricted != tied_human.get_species())
+				allowed_actions -= act
```
```suggestion
		if(act.species_restricted && act.species_restricted != tied_human.get_species())
			allowed_actions -= act
```

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm:
```diff
@@ -136,6 +136,10 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 	// List all allowed action types for AI to consider
 	var/list/allowed_actions = action_whitelist || (GLOB.AI_actions.Copy() - action_blacklist)
+	for(var/datum/ai_action/act in allowed_actions)
+		if(act.species_restricted)
```
not needed

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm:
```diff
@@ -136,6 +136,10 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 	var/list/allowed_actions = action_whitelist || (GLOB.AI_actions.Copy() - action_blacklist)
+	for(var/datum/ai_action/act in allowed_actions)
+		if(act.species_restricted)
+			if(act.species_restricted != tied_human.get_species())
```
store the get_species() call so we can avoid calling it N times

Contrabang reviewed (commented):
some suggestions

Zonespace27 commented on code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm:
```diff
@@ -136,6 +136,10 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 
 	// List all allowed action types for AI to consider
 	var/list/allowed_actions = action_whitelist || (GLOB.AI_actions.Copy() - action_blacklist)
+	for(var/datum/ai_action/act in allowed_actions)
```
as anything

The32bitguy committed:
Merge remote-tracking branch 'upstream/master' into cov-cqc

The32bitguy committed:
Requested Changes, Fixes Skills.

Implements all requested changes from the two reviewers, including making prob(70) not a magic number.

SKILL_MELEE is a typo of SKILL_MELEE_WEAPONS and caused a runtime because it added a skill that did not exist to cAi skill lists. The define did not exist either.

DexterDude reviewed (approved):
awesome
